### PR TITLE
[topic-gpio] Convert GPIO mesh on/off samples to the new GPIO API

### PR DESCRIPTION
**Pull request by carlescufi**
Tested on nrf52840_pca10056

### CODE REVIEW

**Comment by carlescufi** [comment]
@vikrant8051 can you please review and test?

File: samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/main.c [hunk 1]
```diff
@@ -114,30 +109,12 @@ void update_led_gpio(void)
...
-	if (ctl->light->current) {
-		/* LED1 On */
-		gpio_pin_write(led_device[0], DT_ALIAS_LED0_GPIOS_PIN, 0);
-	} else {
-		/* LED1 Off */
-		gpio_pin_write(led_device[0], DT_ALIAS_LED0_GPIOS_PIN, 1);
-	}
-
+	gpio_pin_set(led_device[0], DT_ALIAS_LED0_GPIOS_PIN,
+		       ctl->light->current);
```
Comment (anangl) [inline samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/main.c]: imperfect indentation

**Review by anangl — APPROVED** [review]

**Comment by vikrant8051** [comment]
Ok. I will test

File: samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/main.c (hunk 1 above)
Comment (carlescufi) [reply samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/main.c]: Fixed, thank you.

**Review by pabigot — APPROVED** [review]

File: samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/main.c [hunk 2]
```diff
@@ -94,13 +94,8 @@ static void light_default_status_init(void)
...
 #ifndef ONE_LED_ONE_BUTTON_BOARD
-	if (vnd_user_data.current == STATE_ON) {
-		/* LED2 On */
-		gpio_pin_write(led_device[1], DT_ALIAS_LED1_GPIOS_PIN, 0);
-	} else {
-		/* LED2 Off */
-		gpio_pin_write(led_device[1], DT_ALIAS_LED1_GPIOS_PIN, 1);
-	}
+	gpio_pin_set(led_device[1], DT_ALIAS_LED1_GPIOS_PIN,
+		     vnd_user_data.current == STATE_ON);
```
Comment (mnkp) [inline samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/main.c]: We could write it as `vnd_user_data.current == STATE_ON ? 1 : 0` to be Misra-C compliant. A few more `gpio_pin_set` calls may need to be updated. Non blocking.

**Review by mnkp — APPROVED** [review]

File: samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/app_gpio.c [hunk 3]
```diff
@@ -29,72 +29,56 @@ void app_gpio_init(void)
...
-			    GPIO_PUD_PULL_UP |
-			    GPIO_INT_DEBOUNCE | GPIO_INT_ACTIVE_LOW));
-	gpio_init_callback(&button_cb[3],
-			   button_pressed,
+			   GPIO_INPUT | GPIO_INT_DEBOUNCE |
+			   DT_ALIAS_SW3_GPIOS_FLAGS);
+	gpio_init_callback(&button_cb[3], button_pressed,
 			   BIT(DT_ALIAS_SW3_GPIOS_PIN));
 	gpio_add_callback(button_device[3], &button_cb[3]);
-	gpio_pin_enable_callback(button_device[3], DT_ALIAS_SW3_GPIOS_PIN);
```
Comment (MaureenHelm) [inline samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/app_gpio.c]: are you missing calls to `gpio_pin_interrupt_configure` for sw0 through sw3?

Comment (carlescufi) [reply samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/app_gpio.c]: You are indeed right, fixed.

**Comment by carlescufi** [comment]
@MaureenHelm this should be ready now

**Review by MaureenHelm — APPROVED** [review]

File: samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/app_gpio.c (hunk 3 above)
Comment (mnkp) [reply samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/app_gpio.c]: Well spotted!

**Pull request merged**